AudiusProject/audius-protocol
# Read subscriptions from discovery in notifications system in identity

### Description
Read subscriptions from discovery DB in identity behind the `READ_SUBSCRIBERS_FROM_DISCOVERY_ENABLED` flag. If flag is off: reads from `Subscriptions` in identity (current behavior)

### Tests
Tested against staging DB with debug log statements:
- Turn flag on
- Subscribe account 1 to account 2 and vice versa. Verify subscriptions with API
- On each account:
    - Upload a track
    - Watch logs to verify discprov path is being hit
    - Verify the other account received a notification
![image](https://user-images.githubusercontent.com/25782182/201453254-75bf1a7e-0467-4415-82c1-3b06be1d9fe4.png)
- Turn flag off
- Same flow as above. Verify parity.

### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?
Fails silently with error in logs. No create notifications are sent.
